http.go

```diff
@@ -64,6 +64,64 @@ func (k *Kasse) PostLoginPage(res http.ResponseWriter, req *http.Request) {
 	http.Redirect(res, req, redirect, http.StatusFound)
 }
 
+// GetNewUserPage renders the page to create a new user.
+func (k *Kasse) GetNewUserPage(res http.ResponseWriter, req *http.Request) {
+	res.Header().Set("Content-Type", "text/html")
+
+	if err := ExecuteTemplate(res, TemplateInput{Title: "Create new user", Body: "newUser.html"}); err != nil {
+		k.log.Println("Could not render template:", err)
+		http.Error(res, "Internal error", 500)
+		return
+	}
+}
+
+// PostNewUserPage receives a POST request with username and password and tries
+// to create a new user. It will redirect to the first Flashvalue in the
+// session on success, or to / if none is set and save the authenticated user
+// in the session.
+func (k *Kasse) PostNewUserPage(res http.ResponseWriter, req *http.Request) {
+	username := req.FormValue("username")
+	password := []byte(req.FormValue("password"))
+
+	if username == "" || len(password) == 0 {
+		// TODO: Write own Error function, that uses a template for better
+		// looking error pages. Also, redirect.
+		http.Error(res, "Neither username nor password can be empty", http.StatusBadRequest)
+		return
+	}
+
+	user, err := k.RegisterUser(username, password)
+	if err != nil && err != ErrUserExists {
+		k.log.Println("Something went wrong:", err)
+		// TODO: Write own Error function, that uses a template for better
+		// looking error pages. Also, redirect.
+		http.Error(res, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	if err == ErrUserExists {
+		k.log.Println("User already exists.", err)
```

Comment:
[https://github.com/nnev/kasse/blob/master/main.go#L161](This would appear in) the log as "User already exists. username already taken". Just print err.